src/server_manager/electron_app/index.ts

```diff
@@ -237,3 +102,101 @@ function getWebAppUrl() {
   console.log('Launching web app from ' + webAppUrlString);
   return webAppUrlString;
 }
+
+function main() {
+  // prevent window being garbage collected
+  let mainWindow: Electron.BrowserWindow;
+
+  // Mark secure to avoid mixed content warnings when loading DigitalOcean pages via https://.
+  electron.protocol.registerStandardSchemes(['outline'], {secure: true});
+
+  const isSecondInstance = app.makeSingleInstance((argv, workingDirectory) => {
+    // Someone tried to run a second instance, we should focus our window.
+    if (mainWindow) {
+      if (mainWindow.isMinimized()) {
+        mainWindow.restore();
+      }
+      mainWindow.focus();
+    }
+  });
+
+  if (isSecondInstance) {
+    app.quit();
+  }
+
+  app.on('ready', () => {
+    const menuTemplate = menu.getMenuTemplate(debugMode);
+    if (menuTemplate.length > 0) {
+      electron.Menu.setApplicationMenu(electron.Menu.buildFromTemplate(menuTemplate));
+    }
+
+    // Register a custom protocol so we can use absolute paths in the web app.
+    // This also acts as a kind of chroot for the web app, so it cannot access
+    // the user's filesystem (important for the DigitalOcean pages we customise).
```

Comment:
Do we still customize any DigitalOcean pages?

Reply:
Nope. I removed that comment